rules/import_middleman.bzl

```diff
@@ -264,15 +264,9 @@ def _file_collector_rule_impl(ctx):
     )
 
     additional_providers = []
-    if not virtualize_frameworks:
-        dep_cc_infos = [dep[CcInfo] for dep in ctx.attr.deps if CcInfo in dep]
-        cc_info = cc_common.merge_cc_infos(direct_cc_infos = [], cc_infos = dep_cc_infos)
-        additional_providers.append(cc_info)
-    else:
-        cc_info = objc_provider_utils.merge_cc_info_providers(
-            cc_info_providers = [dep[CcInfo] for dep in ctx.attr.deps if CcInfo in dep],
-        )
-        additional_providers.append(cc_info)
```

Comment:
This custom merging logic caused a problematic `modulemap` is generated. I'm not sure why we are merging `CcInfo` differently depending on whether VFS is used, but all CI jobs are passed.

Reply:
We should be able to though right? If we can't rely on CI being 🟢 here then I think we should look into it and make CI start checking for whatever is necessary?

Reply:
Yeah, I have the repro example in the issue #569 and the repro code [here](https://github.com/qyang-nj/BazelPlayground/tree/main/Issue0). I thought about adding it to CI tests, but realized we don't have any tests run under `--compilation_mode=opt`. Maybe we can add that.

Reply:
SGTM! Yeah it'd be nice to improve the coverage for the things we can, I can dig more into this early next week.